### ISSUE TITLE
Image.create_from_data has no sensible way of passing multi-byte samples 

### ISSUE THREAD

**Issue by kelvin13** [godotengine/godot#49700]
### Godot version

3.3

### System information

N/A

### Issue description

When using [`Image`](https://docs.godotengine.org/en/stable/classes/class_image.html).[`create_from_data`](https://docs.godotengine.org/en/stable/classes/class_image.html#class-image-method-create-from-data) with a multibyte format such as [`FORMAT_RGBF`](https://docs.godotengine.org/en/stable/classes/class_image.html#enum-image-format), there is no sensible way to pass the required 32-bit floating point samples as the expected `PoolByteArray` argument. 

The byte-order of the `Float32` samples is not specified anywhere, which means we cannot safely reinterpret the original `Float32` samples as a raw buffer of `UInt8` values, and pass that to  [`Image`](https://docs.godotengine.org/en/stable/classes/class_image.html).[`create_from_data`](https://docs.godotengine.org/en/stable/classes/class_image.html#class-image-method-create-from-data).

### Steps to reproduce

N/A

**Comment by timothyqiu** [comment]
It's native byte order according to the source code:

https://github.com/godotengine/godot/blob/7610409b8a14b8499763efa76578795c755a846d/core/image.cpp#L2613-L2618

**Comment by Calinou** [comment]
> But it forces us to choose between big-endian and little-endian.

Wasn't big-endian serialization removed from `master`?

**Comment by timothyqiu** [comment]
@Calinou It's still there on the current `master`.

https://github.com/godotengine/godot/blob/8ffb631cdb740540ae74edaa5cef8a04d263ccee/core/io/stream_peer.cpp#L181-L191

Yeah, in practice, one can use little endian where native endian is needed because Godot only runs on platforms that is little endian native. But it's wrong semantically.

**Comment by Xrayez** [comment]
Yeah, some image formats can be written and read as little/big endian, so likely both should be supported even today.

See also https://people.cs.umass.edu/~verts/cs32/endian.html.

**Comment by kelvin13** [comment]
> It's native byte order according to the source code:
> 
> https://github.com/godotengine/godot/blob/7610409b8a14b8499763efa76578795c755a846d/core/image.cpp#L2613-L2618

does this mean a `reinterpret_cast` can be used from the GDNative side, and we can assume the GDNative library uses the same endianness as the engine binary? I am using Swift nativescripts, and Swift has utilities for dealing with endianness.

**Comment by timothyqiu** [comment]
> does this mean a `reinterpret_cast` can be used from the GDNative side, and we can assume the GDNative library uses the same endianness as the engine binary?

I think so. Not sure how Swift works, but in C++ you can `reinterpret_cast` a pointer to `float`s into a pointer to bytes, and use it to fill your `PoolByteArray`.
